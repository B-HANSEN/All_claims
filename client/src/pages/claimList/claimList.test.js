import React from 'react';
import { render, screen } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import '@testing-library/jest-dom';
import ClaimList from '.';

beforeEach(() => {
	render(<ClaimList />);
});

describe('Claim List Page tests', () => {
	it('should match snapshot', () => {
		const renderer = new ShallowRenderer();
		const result = renderer.render(<ClaimList />);
		expect(result).toMatchSnapshot();
	});

	it('should contain heading, label and button texts', () => {
		const heading = screen.getByText(/overview claims/i);
		const searchLabel = screen.getByText(
			/search by id, holder name or policy/i
		);
		const query = screen.getByPlaceholderText(/enter query here.../i);
		const filter = screen.getByText(/show all/i);
		const status = screen.getByText(/filter by claim status/i);
		const createNew = screen.getByRole('button', {
			name: /create a new claim/i,
		});
		const admin = screen.getByRole('button', {
			name: /admin section/i,
		});

		const submit = screen.getByRole('button', {
			name: /submit/i,
		});
		const reset = screen.getByRole('button', {
			name: /reset/i,
		});

		expect(heading).toBeInTheDocument();
		expect(searchLabel).toBeInTheDocument();
		expect(query).toBeInTheDocument();
		expect(filter).toBeInTheDocument();
		expect(status).toBeInTheDocument();
		expect(createNew).toBeInTheDocument();
		expect(createNew).toBeEnabled();
		expect(admin).toBeInTheDocument();
		expect(admin).toBeEnabled();
		expect(submit).toBeInTheDocument();
		expect(submit).toBeEnabled();
		expect(reset).toBeInTheDocument();
		expect(reset).toBeEnabled();
	});
});
