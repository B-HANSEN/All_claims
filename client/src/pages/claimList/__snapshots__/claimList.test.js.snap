// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Claim List Page tests should match snapshot 1`] = `
<div
  className="ClaimList"
>
  <Header />
  <section
    className="ClaimList__MainSection"
  >
    <h1>
      Overview claims
    </h1>
    <section
      className="ClaimList__Filter"
    >
      <Search
        setValue={[Function]}
      />
      <StatusFilter
        filter="ShowAll"
        setFilter={[Function]}
      />
    </section>
    <Button
      className="mb-2"
      href="/?admin=true"
      size="lg"
      variant="outline-success"
    >
      Admin Section
    </Button>
     
    <Button
      className="mb-2"
      href="/create-claim"
      size="lg"
      variant="outline-success"
    >
      Create a new claim
    </Button>
    <section
      className="ClaimList__Table"
    >
      <ClaimsTable
        filteredClaims={Array []}
      />
    </section>
  </section>
</div>
`;
